### ISSUE TITLE
Provide more nodes in artificial neural networks

### ISSUE THREAD

**Issue by chaoming0625** [PKU-NIP-Lab/BrainPy#92]
Please:

- [x] Check for duplicate requests.
- [x] Describe your goal, and if possible provide a code snippet with a motivating example.

Currently, BrainPy provides excellent nodes in reservoir computing and recurrent neural networks. However, there are much more layers or modles which are useful when constructing dynamics training models. Such as Convolution kernels, transformers, batch normarlizations,etc.

**Comment by chaoming0625** [comment]
See recent advances: 

- #138 
- #142

**Comment by chaoming0625** [comment]
Recent advances:

- #162 
- #161
